KubeArmor/core/kubeUpdate.go

```diff
@@ -456,36 +457,42 @@ func (dm *KubeArmorDaemon) WatchK8sPods() {
 				pod.Metadata["namespaceName"] = event.Object.ObjectMeta.Namespace
 				pod.Metadata["podName"] = event.Object.ObjectMeta.Name
 
+				pod.Labels = map[string]string{}
+				for k, v := range event.Object.Labels {
+					if k == "pod-template-generation" {
+						continue
+					}
+
+					if k == "controller-revision-hash" {
+						continue
+					}
+					pod.Labels[k] = v
+				}
+
 				if len(event.Object.ObjectMeta.OwnerReferences) > 0 {
+					// pod is owned by a replicaset, daemonset etc thus we use parent object's name
+					pod.Metadata["ownerName"] = event.Object.ObjectMeta.OwnerReferences[0].Name
+
 					if event.Object.ObjectMeta.OwnerReferences[0].Kind == "ReplicaSet" {
 						deploymentName := K8s.GetDeploymentNameControllingReplicaSet(pod.Metadata["namespaceName"], event.Object.ObjectMeta.OwnerReferences[0].Name)
 						if deploymentName != "" {
 							pod.Metadata["deploymentName"] = deploymentName
 						}
+
+						// if it belongs to a replicaset, we also remove the pod template hash
+						podTemplateHashSuffix := fmt.Sprintf("-%s", pod.Labels["pod-template-hash"])
+						pod.Metadata["ownerName"] = strings.TrimSuffix(event.Object.ObjectMeta.OwnerReferences[0].Name, podTemplateHashSuffix)
```

Comment:
When I checked my sample deployment, I have a pod that was instantiated with name `wordpress-54dd4449df-wq7ql` ... but the `pod-template-hash` was only `54dd4449df`. `TrimSuffix` won't be able to trim in this case.

```
vagrant@kubearmor-dev:~/KubeArmor/tests$ kubectl describe pod -n wordpress-mysql wordpress-54dd4449df-wq7ql
Name:         wordpress-54dd4449df-wq7ql
Namespace:    wordpress-mysql
Priority:     0
Node:         kubearmor-dev/10.0.2.15
Start Time:   Wed, 06 Jul 2022 07:25:24 +0000
Labels:       app=wordpress
              pod-template-hash=54dd4449df
Annotations:  container.apparmor.security.beta.kubernetes.io/wordpress: localhost/kubearmor-wordpress-mysql-wordpress
              kubearmor-policy: enabled
Status:       Running
IP:           10.0.0.69
IPs:
  IP:           10.0.0.69
Controlled By:  ReplicaSet/wordpress-54dd4449df
Containers:
  wordpress:
    Container ID:   docker://b503dfed67c4ec54c01b9f91437a089bf44913092bd1da66b38f43c1a7f94b32
    Image:          wordpress:4.8-apache
    Image ID:       docker-pullable://wordpress@sha256:6216f64ab88fc51d311e38c7f69ca3f9aaba621492b4f1fa93ddf63093768845
    Port:           80/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Wed, 06 Jul 2022 07:25:32 +0000
    Ready:          True
    Restart Count:  0
    Environment:
      WORDPRESS_DB_HOST:      mysql
      WORDPRESS_DB_PASSWORD:  root-password
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-b4j9b (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  kube-api-access-b4j9b:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:                      <none>
```

Reply:
:white_check_mark: